### ISSUE TITLE
Json::LogicError thrown when switching workspaces

### ISSUE THREAD

**Issue by misopeth** [drmgc/i3ipcpp#35]
I wrote a daemon-like program which subscribes to `i3::ET_WINDOW` and `i3::ET_WORKSPACE`. In the last months I noticed that it often crashes with the following dump:
```
terminate called after throwing an instance of 'Json::LogicError'
  what():  LargestInt out of UInt range
```
Could it be because of new parameters that `i3` added to windows for which `i3ipc++` doesn't properly set the parsing? I'm thinking specifically about `window_icon_padding`, or `current_border_width` (I'm not sure this is new though).

**Comment by misopeth** [comment]
Actually, I noticed that `i3ipc::rect_t` has `x` and `y` of type `uint32_t`, but when I run
```sh
i3-msg -t get_tree
```
in a terminal, `floating_nodes` have `"rect": { "x": -2, ...}`, which I think is more likely to be the issue.

**Comment by drmgc** [comment]
Closed by bd71b09, 83f4e80
